pkg/controller/podautoscaler/horizontal.go

```diff
@@ -271,15 +271,15 @@ func (a *HorizontalController) reconcileAutoscaler(hpa *autoscaling.HorizontalPo
 	rescaleReason := ""
 	timestamp := time.Now()
 
-	if currentReplicas > hpa.Spec.MaxReplicas {
+	if scale.Spec.Replicas == 0 {
+		// Autoscaling is disabled for this resource
+		desiredReplicas = 0
+	} else if currentReplicas > hpa.Spec.MaxReplicas {
 		rescaleReason = "Current number of replicas above Spec.MaxReplicas"
 		desiredReplicas = hpa.Spec.MaxReplicas
 	} else if hpa.Spec.MinReplicas != nil && currentReplicas < *hpa.Spec.MinReplicas {
 		rescaleReason = "Current number of replicas below Spec.MinReplicas"
 		desiredReplicas = *hpa.Spec.MinReplicas
-	} else if currentReplicas == 0 {
```

Comment:
We should do something if there is no replicas present. It can happen that HPA controller is fired before RC/RS/Deployment controller causing current replicas to be zero while spec target to be non zero. In this case we should either skip HPA loop or set some safe desired replica count so 1 or spec.replicas.